pkg/terraform/format.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package terraform
+
+import (
+	"context"
+	"fmt"
+	"io"
+)
+
+// FormatOptions are the set of options for running terraform format.
+type FormatOptions struct {
+	Check     *bool
+	Diff      *bool
+	List      *bool
+	NoColor   *bool
+	Recursive *bool
+	Write     *bool
+}
+
+// formatArgsFromOptions generated the terrafrom format arguments from the provided options.
+func formatArgsFromOptions(opts *FormatOptions) []string {
+	args := make([]string, 0, 6) // 6 potential args to be added
+
+	if opts == nil {
+		return args
+	}
+
+	if opts.Check != nil && *opts.Check {
```

Comment:
It might be helpful to make this a generic function like `boolVal`

```go
func boolVal(b *bool) bool {
  if v == nil {
    return false
  }
  return *v
}
```

of if you wanted to be really fancy:

```go
func boolVal(b *bool) (val, ok bool) {
  if v == nil {
    return
  }
  return *v, true
}
```

Reply:
I prefer option 1, will update